### PR TITLE
[ML][Data Frame] Add deduced mappings to _preview response payload

### PR DESCRIPTION
**Pull request by benwtrent**
This updates the `_preview` response payload to include the deduced mappings. The "mapping" format returned from `Pivot#deduceMappings` has a non-standard format of `{"field1": "field1Type", "field2.subfield": "field2.subfieldType"}`. So, I opted to transform it so that it is in an appropriate format of 
```
"properties": {
   "field1": {
       "type": "field1Type"
   },
   "field2.subfield": {
       "type": "field2.subfieldType"
   }
}
```

Related to: elastic/kibana#39250

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PreviewDataFrameTransformAction.java [hunk 1]
```diff
@@ -166,18 +174,56 @@ public void setDocs(List<Map<String, Object>> docs) {
...
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeInt(docs.size());
             for (Map<String, Object> doc : docs) {
                 out.writeMapWithConsistentOrder(doc);
             }
+            if (out.getVersion().onOrAfter(Version.V_7_3_0)) {
+                out.writeMap(mappings);
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PreviewDataFrameTransformAction.java]: it seems mappings can be `null`, so would it make sense to use optional?

**Review by hendrikmuhs — COMMENTED** [review]
1 question, looks good otherwise.

Wondering about check in: do not we have to use Version.Current and change it to 7.3.0 in a followup to not break BWC?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PreviewDataFrameTransformAction.java (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PreviewDataFrameTransformAction.java]: @hendrikmuhs `writeMap` and `readMap` when it is just a `Map<String,Object>` handles `null` values internally. 

There are also no BWC tests yet...still waiting on PR: https://github.com/elastic/elasticsearch/pull/43506

Comment (hendrikmuhs) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PreviewDataFrameTransformAction.java]: ok checked it, implicit magic, well...

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/1
run elasticsearch-ci/packaging-sample

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**